colossus/src/main/scala/colossus/protocols/http/filters/HttpCustomFilters.scala

```diff
@@ -0,0 +1,39 @@
+package colossus.protocols.http.filters
+
+import colossus.protocols.http.{ContentEncoding, Http, HttpBody, HttpHeaders}
+import colossus.service.Filter
+import colossus.service.GenRequestHandler.PartialHandler
+import colossus.util.{GzipCompressor, ZCompressor}
+
+
+object HttpCustomFilters {
+
+  class CompressionFilter(bufferedKb: Int = 10) extends Filter[Http] {
+
+    override def apply(next: PartialHandler[Http]): PartialHandler[Http] = {
+      case request =>
```

Comment:
Any way we can get rid of the duplicate logic between this and the streaming version?